### DISCUSSION TITLE
Using DeepXDE to solve PDE's in a loss function as part of a training loop in tensorflow

### DISCUSSION THREAD

**Discussion by flipflop27** [lululxvi/deepxde#1660] [General]
Apologies if this isn't worded correctly as I am still relatively new to machine learning and the DeepXDE library.
I am trying to formulate a neural network that can estimate pressures and flowrates in a 1D pipeline as a function of time and space. The PDEs that govern transient flow are shown below:
![image](https://github.com/lululxvi/deepxde/assets/150153530/c28b17f1-b33f-4554-a4c5-b9c7d60e3111)

My question is: is it possible to use DeepXDE to compute the PDE's within my existing training loop to try and reduce both F1 and F2 to as close to 0 as possible for every iteration in my training loop. The outputs of the neural network are q_hat and h_hat which are both functions of x and t.

**Comment by lululxvi** [comment]
Yes. Just learn how to use DeepXDE.
